levels/lovelace_tower/events.js

```diff
@@ -1,353 +1,162 @@
-      // destroyObject("catacombs_barrier");
-      openDoor("catacombs_barrier");
-      unhackObject("statue_credentials");
-    }
+  // If No:
+  // [] Doors 1/2/3 are locked respectively with Operator observation: "I need to complete the next objective first."
 
-    // Adjust object dimensions
-    world.forEachEntities(
-      ({ instance }) => instance.layer === "upper",
-      (object) => {
-        // To force "upper" layer objects to render above objects
-        // they're placed on top of, we must hack their physics
-        // body size.
-        //
-        // To do this, we're going to add a hard coded 1000 pixels
-        // to the height of these objects. This is not perfect,
-        // but should function fine for this mission.
-        object.sprite.body.height += 1000;
+  // If Yes:
+  // [] Door 1/2/3 is open (door objects should be hidden from view on the map)
 
-        // These sprites will all have their bodies disabled.
-        // This new tall sprite body will impede a lot of player
-        // movement space otherwise.
-        object.sprite.body.enable = false;
-      }
-    );
-  }
+  // Has player completed all four Tower objectives? (obj4Complete)
 
-// Handles object interactions
-  if (event.name === "playerDidInteract") {
-    console.log(`Interacting with ${event.target.key}`);
+  // If No:
+  // [x] Secret Library Door is locked and triggers library-door.pug dialogue
+  // [x] Groundskeeper says dialogue 1
 
-    if (event.target.spellable) runSpell(event);
-    if (event.target.notify) runObjectNotification(event);
-    if (event.target.npc) runNpcChecks(event);
-    if (event.target.item) runAddItem(event);
-  }
+  // If Yes:
+  // [] Spell in inventory
+  // [] Sparkle animation appears on secret Library door
+  // [x] Groundskeeper says dialogue 2
+  // [] Library door is unlocked and player can teleport to library map
 
-// Handles objective completions & failures
-  if (
-    event.name === "objectiveCompleted" ||
-    event.name === "objectiveCompletedAgain"
-  ) {
-    runObjectiveEffects(event);
-  }
+  // Has the player completed the final objective? (obj5Complete)
 
-  if (event.name === "objectiveFailed") {
-    runUpdateMagicScore(event);
-  }
+  // If No:
+  // [] Exit door from library is locked with Operator observation
 
-// Handles Trigger Areas.
-// This functionality should be updated so that choosing the house in the House Ceremony 
-// triggers interrupting professor dialogue and camera pan
-// This should also trigger the opening of Lovelace Tower
-  if (event.name === "triggerAreaWasEntered") {
-    if (
-      event.target.key === "triggerGroundskeeperConversation" &&
-      !worldState.insidePerimeter.groundskeeper_introduction
-    ) {
-      world.startConversation("groundskeeper", "groundskeeper.png");
-    }
-  }
+  // If Yes:
+  // [] Sparkle animation appears on exit door and is interactable
+  // [] Previous entry door back into Tower becomes locked
```

Comment:
Is this all the pre-existing comment stuff?

Reply:
Yep! I believe @nmadadair has been using them to keep track of things, if I'm not mistaken

Reply:
Comments are mine, but I'm deleting them as the code gets written. Regarding old inside_perimeter event.js code, we've deleted it all and are starting from scratch for both new levels.